client/campaign_message_configuration/README.md

```diff
@@ -0,0 +1,79 @@
+# Campaign Message Configuration
+
+Store campaign message configurations for their respective campaigns. This allows users to switch between campaigns while keeping each of the message configurations and restoring them when the user returns to the original campaign. New campaigns can either be created with customizable default
+settings or from the existing configuration (see the "Transfer Settings" option).
+
+## Getting Started
+
+These instructions will get you a copy of the plugin up and running on your local machine for development and testing purposes. 
+
+### Prerequisites
+
+What you need to install the software:
+
+```
+Linux or Windows 
+King Phisher Minimum Required Version = 1.10.0 
+```
+
+### Installing 
+
+- [ ] Download plugin 
+- [ ] Move plugin to: <code>/opt/king-phisher-plugins/client/</code>
+- [ ] Configure plugin<code>(refer to Deployment & Configuration below)</code> 
+
+## Running the tests
+
+In order to test if `campaign_message_configuration.py` is operational, we must change campaigns then return back to the original campaign. To create a new campaign, navigate to the King Phisher Client. Once inside of the client we need to open the campaigns menu, to do this go to `File -> Open Campaign -> New Campaign. Once you have generated a new campaign, revert back to the original campaign that you created and check that your settings, etc.. If all campaign information is being saved and properly carried over then the plugin is working properly!  
+
+>It is recommended that end users implementing this plugin run the King Phisher Client in debug mode before testing the plugin. `DEBUG` is accomplished by inserting `-L DEBUG` onto the execution command: 
+```
+# CLIENT EXAMPLE 
+testing@testVM:/opt/king-phisher$ ./KingPhisher -L DEBUG
+```
+```
+# SERVER EXAMPLE 
+testing@testVM:/opt/king-phisher$ sudo ./KingPhisherServer server_config.yml -L DEBUG
+```
+
+Once `DEBUG` logging has been enabled, send out the phishing attempt and keep an eye on the `DEBUG` logging. Once target opens phishing URL, the redirect should dispatch onto a new tab within the browser window. If you see any `WARNING` or `CRITICAL` within the log, check your configurations and begin the README process again. 
+
+## Deployment & Configuration 
+
+On line 40 of `campaign_message_configuration.py` is the start of `Client Options`. If the client does not specify the `Client Options`, the client will receive default values for those options. 
+
+To implement your options to <code>Client Options</code>, navigate to <code>/king-phisher/server_config.yml</code> 
+
+Once server_config.yml is open, navigate to <code>#plugins</code> within the file. Here we will implement our own options:
+>Note that proper spacing is critical to the server_config.yml file.
+```
+#EXAMPLE

```

Comment:
again this only true for server plugins..

Once a client plugin is loaded and enabled you can change its setting through the client
Edit > Preferences
In King Phusher Configuartion window that opens select the Plugins tab